src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs

```diff
@@ -60,10 +60,13 @@ public static Task<ProxyResult> StartAsync(out int port, bool requireAuth)
                 await getAndReadRequest().ConfigureAwait(false);
 
                 // If we're expecting credentials, look for them, and if we didn't get them, send back 
-                // a response and process a new request.
+                // a 407 response. Optionally, process a new request that would expect credentials.
                 if (requireAuth && !headers.ContainsKey("Proxy-Authorization"))
                 {
-                    Task<Socket> secondListen = listener.AcceptSocketAsync();
+                    if (expectCreds)
+                    {
+                        Task<Socket> secondListen = listener.AcceptSocketAsync();
+                    }
```

Comment:
Do we need this at all?   I think I inadvertently left this line in with some changes before, and it looks like it's unnecessary and could even cause problems in the test.